src/crimson/net/Protocol.cc

```diff
@@ -96,6 +96,48 @@ void Protocol::notify_keepalive_ack(utime_t _keepalive_ack)
   write_event();
 }
 
+void Protocol::requeue_sent()
+{
+  assert(write_state != write_state_t::open);
+  if (conn.sent.empty()) {
+    return;
+  }
+
+  conn.out_seq -= conn.sent.size();
+  logger().debug("{} requeue {} items, revert out_seq to {}",
+                 conn, conn.sent.size(), conn.out_seq);
+  for (MessageRef& msg : conn.sent) {
```

Comment:
Why we don't call `SocketConnection::requeue_sent()`? Is that because of inserting at the front of `out_q`?

Anyway, we still could do the move in single iteration:

```cpp
  conn.out_seq -= conn.sent.size();
  while (!conn.sent.empty()) {
    auto& m = conn.sent.back();
    m->clear_payload();
    m->set_seq(0);
    conn.out_q.push_front(std::move(m));
    conn.sent.pop_back();
  }
```
An alternative might be to use `std::move` from the `algorithm` library.

Reply:
Oops, forgot to remove the duplicated `SocketConnection::requeue_sent()`, it was implemented for v1 protocol.

Currently the related functions of stateful write are in `Protocol` class.